api/v1beta1/common_types.go

```diff
@@ -126,6 +126,12 @@ const (
 	// any changes to the actual object because it is a dry run) and the topology controller
 	// will receive the resulting object.
 	TopologyDryRunAnnotation = "topology.cluster.x-k8s.io/dry-run"
+
+	// ReplicasManagedByExternalAutoscalerAnnotation is an annotation that indicates an external autoscaler manages infra scaling.
+	// The practical effect of this is that the capi "replica" count should be passively derived from the number of observed infra machines,
+	// instead of being a source of truth for eventual consistency.
+	// This annotation can be used to inform MachinePool status during in-progress scaling scenarios.
+	ReplicasManagedByExternalAutoscalerAnnotation = "cluster.x-k8s.io/replicas-managed-by-external-autoscaler"
```

Comment:
I like the brevity of this suggested change, but FYI this has been discussed in a previous thread. Any thoughts @enxebre @AverageMarcus @CecileRobertMichon ?

Reply:
Well, it could be managed by some other non-capi scaling interface. But perhaps "autoscaler" is a valid term to encompass all such things.

It would seem that if we want to put some of the semantic significance of this annotation in the value, then we shouldn't define this as a "boolean" value any longer. We would want the exact key and val to match. I think that's fine here, I'm less concerned with exactly what this annotation looks like and more concerned with defining it once so that it is "well-known".

Reply:
I think this might be a typical case of let's not design ourselves into a corner (if we think there could be possible values going forward). 

What I could imagine is that for some reason it might be somehow relevant who exactly the autoscaler is so folks could use more specific values at some point without us having to deprecate the annotation and creating a new one.

Reply:
Yep, fair enough. I'll go ahead and push @vincepri's suggestion up so we can evaluate what that looks like (and it'll help me scaffold out some docs)

Reply:
I'm good with that suggestion as well - as long as the providers set the value to reflect what the replicas are managed by it should provide enough information. I think the risk of this not merging and providers each coming up with their own version of this change is the important thing here.

Reply:
Pinging @mweibel too as they have a PR to add this functionality to CAPA (https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3683)

Reply:
thanks @AverageMarcus I didn't watch this PR for some reason and didn't see this discussion. 
Change of annotation sounds fine to me :+1: Will adjust my PR accordingly.